src/views/v2/commerce/checkout/gateways.php

```diff
@@ -26,8 +26,13 @@
  * @var int                $gateways_connected [Global] The number of connected gateways.
  */
 
-// Bail if user needs to login, the cart is empty or if there are no active gateways.
-if ( $must_login || empty( $items ) || ! tribe_is_truthy( $gateways_active ) ) {
+// Bail if the cart is empty or if there are no active gateways.
+if ( empty( $items ) || ! tribe_is_truthy( $gateways_active ) ) {
+	return;
+}
+
+// Bail if user needs to login, but is not logged in.
+if ( ! is_user_logged_in() && $must_login ) {
```

Comment:
A micro optimization, but let's move the more expensive item second:

```suggestion
if ( $must_login && ! is_user_logged_in() ) {
```